lib/taskana-core-test/src/test/java/acceptance/classification/delete/DeleteClassificationAccTest.java

```diff
@@ -0,0 +1,290 @@
+import static org.assertj.core.api.Assertions.catchThrowableOfType;
+import static pro.taskana.common.api.SharedConstants.MASTER_DOMAIN;
+
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import pro.taskana.classification.api.ClassificationService;
+import pro.taskana.classification.api.exceptions.ClassificationInUseException;
+import pro.taskana.classification.api.exceptions.ClassificationNotFoundException;
+import pro.taskana.classification.api.models.Classification;
+import pro.taskana.common.api.TaskanaRole;
+import pro.taskana.common.api.exceptions.MismatchedRoleException;
+import pro.taskana.task.api.TaskService;
+import pro.taskana.task.api.models.Attachment;
+import pro.taskana.testapi.DefaultTestEntities;
+import pro.taskana.testapi.TaskanaInject;
+import pro.taskana.testapi.TaskanaIntegrationTest;
+import pro.taskana.testapi.builder.TaskAttachmentBuilder;
+import pro.taskana.testapi.builder.TaskBuilder;
+import pro.taskana.testapi.builder.WorkbasketAccessItemBuilder;
+import pro.taskana.testapi.security.WithAccessId;
+import pro.taskana.workbasket.api.WorkbasketPermission;
+import pro.taskana.workbasket.api.WorkbasketService;
+import pro.taskana.workbasket.api.models.Workbasket;
+
+@TaskanaIntegrationTest
+class DeleteClassificationAccTest {
+
+  @TaskanaInject ClassificationService classificationService;
+  @TaskanaInject TaskService taskService;
+  @TaskanaInject WorkbasketService workbasketService;
+
+  Workbasket defaultWorkbasket;
+
+  @WithAccessId(user = "businessadmin")
+  @BeforeAll
+  void setup() throws Exception {
+    defaultWorkbasket =
+        DefaultTestEntities.defaultTestWorkbasket().buildAndStore(workbasketService);
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(defaultWorkbasket.getId())
+        .accessId("businessadmin")
+        .permission(WorkbasketPermission.OPEN)
+        .permission(WorkbasketPermission.READ)
+        .permission(WorkbasketPermission.APPEND)
+        .buildAndStore(workbasketService, "admin");
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_NotFindClassificationById_When_DeletingClassificationInDomain() throws Exception {
+    Classification classification =
+        DefaultTestEntities.defaultTestClassification()
+            .domain("DOMAIN_A")
+            .buildAndStore(classificationService);
+
+    classificationService.deleteClassification(classification.getKey(), classification.getDomain());
+    ThrowingCallable call = () -> classificationService.getClassification(classification.getId());
+
+    ClassificationNotFoundException e =
+        catchThrowableOfType(call, ClassificationNotFoundException.class);
+    assertThat(e.getClassificationId()).isEqualTo(classification.getId());
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_ThrowException_When_UsingIdAndUserRoleIsNotAdminOrBusinessAdmin() throws Exception {
```

Comment:
I think this should be named should_ThrowException_When_UsingKeyAndDomainAndUserRoleIsNotAdminOrBusinessAdmin() and the test below should_ThrowException_When_UsingIdAndUserRoleIsNotAdminOrBusinessAdmin() :)

Reply:
You are right, I accidentally swapped the names. This will be fixed :)